src/Temporalio/Client/EnvConfig/ClientEnvConfig.cs

```diff
@@ -0,0 +1,414 @@
+            DataSource? ClientPrivateKey = null)
+        {
+            /// <summary>
+            /// Create a Tls from a dictionary structure.
+            /// </summary>
+            /// <param name="dictionary">The dictionary to convert from.</param>
+            /// <returns>TLS configuration instance.</returns>
+            public static Tls FromDictionary(IReadOnlyDictionary<string, object?> dictionary) => new(
+                    ServerName: dictionary.TryGetValue("server_name", out var serverName) ? (string?)serverName : null,
+                    ServerRootCACert: dictionary.TryGetValue("server_ca_cert", out var serverCaCert) ? DataSource.FromDictionary((IReadOnlyDictionary<string, string>?)serverCaCert) : null,
+                    ClientCert: dictionary.TryGetValue("client_cert", out var clientCert) ? DataSource.FromDictionary((IReadOnlyDictionary<string, string>?)clientCert) : null,
+                    ClientPrivateKey: dictionary.TryGetValue("client_key", out var clientKey) ? DataSource.FromDictionary((IReadOnlyDictionary<string, string>?)clientKey) : null,
+                    Disabled: dictionary.TryGetValue("disabled", out var disabled) ? (bool?)disabled : null);
+
+            /// <summary>
+            /// Create a <see cref="TlsOptions"/> from this configuration.
+            /// </summary>
+            /// <returns>TLS options for a client.</returns>
+            public TlsOptions? ToTlsOptions()
+            {
+                if (Disabled == true)
+                {
+                    return null;
+                }
+
+                return new TlsOptions
+                {
+                    Domain = ServerName,
+                    ServerRootCACert = ServerRootCACert?.Data,
+                    ClientCert = ClientCert?.Data,
+                    ClientPrivateKey = ClientPrivateKey?.Data,
+                };
+            }
+
+            /// <summary>
+            /// Convert to a dictionary structure that can be used for TOML serialization.
+            /// </summary>
+            /// <returns>Dictionary representation of this TLS config.</returns>
+            public Dictionary<string, object?> ToDictionary() =>
+                new()
+                {
+                    ["disabled"] = Disabled,
+                    ["server_name"] = ServerName,
+                    ["server_ca_cert"] = ServerRootCACert?.ToDictionary(),
+                    ["client_cert"] = ClientCert?.ToDictionary(),
+                    ["client_key"] = ClientPrivateKey?.ToDictionary(),
+                };
+        }
+
+        /// <summary>
+        /// Options for loading a specific client configuration profile.
+        /// </summary>
+        public class ProfileLoadOptions : ICloneable
+        {
+            /// <summary>
+            /// Initializes a new instance of the <see cref="ProfileLoadOptions"/> class.
+            /// </summary>
+            public ProfileLoadOptions()
+            {
+            }
```

Comment:
```suggestion
```

Default constructor is implied when there are no other constructors (same for other options class)